src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,59 @@
+package duke.task;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * A wrapper for a duke.task.Deadline which is a duke.task.
+ * Task that needs to be done before a date/time.
+ *
+ * @author Wong Yun Rui Chris
+ */
+public class Deadline extends Task {
+    protected String by;
+    protected LocalDate byLd;
+
+    /**
+     * A public constructor to initialise a duke.task.Deadline duke.task.Task.
+     *
+     * @param description The String description/name of the task
+     * @param by The String describing when the duke.task.Deadline need to be done by
+     */
+    public Deadline(String description, String by, Boolean isDone) {
+        super(description, isDone);
+        this.by = by;
```

Comment:
Maybe make this 'by' variable a little more specific? The byLd is the LocalDate version of it so it would be slightly clearer to be more specific with by.